### ISSUE TITLE
Library Catalog (Encore).js doesn't work outside of Zotero

### ISSUE THREAD

**Issue by Halibutt** [zotero/translation-server-old#16]
Referred from [zotero/translators issue #933](https://github.com/zotero/translators/issues/933)

Recently the National Library of Poland (www.bn.org.pl) [switched to a new OPAC system Encore](http://katalogi.bn.org.pl/iii/encore/?lang=eng). The links work perfectly well in Zotero (FF, Chrome, all the same).  A book is correctly identified as a book, all the fields get imported correctly, it works like a charm. When I export the citation from Zotero using the native Wikipedia exporter, it looks perfectly healthy. 

BUT, it seems the translator does not work outside of the Zotero program, neither on translation-server nor - more importantly for me - in Citoid, the new Wikipedia's citation system based on Zotero's translators. Links that were correctly scraped as books in Zotero either [return weird errors](http://katalogi.bn.org.pl/iii/encore/record/C__Rb1790808__SJemielniak__Ff%253Afacetmediatype%253Aa%253Aa%253AKsi%25C4%2585%25C5%25BCka%253A%253A__P1%252C25__Orightresult__U__X4?lang=pol&suite=cobalt) or are incorrectly identified as web pages, [this book for instance](http://katalogi.bn.org.pl/iii/encore/record/C__Rb3707725__SJemielniak__Orightresult__U__X4?lang=pol&suite=cobalt). 

I [filed a bug at Citoid's Phabricator](https://phabricator.wikimedia.org/T108772), but it seems the problem is on translation server side, not with Citoid. The guys over at the translators' git referred me here. Could anyone tell me how to make this beauty work?

**Comment by wetneb** [comment]
If I am not wrong, this translator is not enabled for translation-server (it would require the "v" code):

<pre>
"browserSupport": "gcsb",
</pre>

Have you tried with the patch proposed in https://github.com/zotero/translators/pull/920? (Actually, Citoid probably already deals with this problem by default…)

**Comment by Halibutt** [comment]
Hmmm... I pinged the Citoid guys to check whether this is the issue, thanks a lot. 
Cheers

**Comment by mvolz** [comment]
Yeah, we tried enabling it. Doesn't work still :( See comment here: https://phabricator.wikimedia.org/T108772#1530977

Re: zotero/translators#920 Some, but not all, of the disabled translators do not work when enabled. Usually it's a problem with getting the resource via the server versus a browser, we have that trouble with citoid as well. This site also throws the native citoid scraper for a loop too.

**Comment by Halibutt** [comment]
Perhaps it's a bug inside Encore itself? I mailed the iii.com guys, hopefully they'll be able to shed some light on it.

**Comment by Halibutt** [comment]
Sadly, no reply from them as of yet.

**Comment by Halibutt** [comment]
Unfortunately, no updates in any of the threads. There is a workaround for the National Library of Poland (their old OPAC system is still online), but all the other libraries out there using Encore are a no-go still.

**Comment by dstillman** [comment]
So this isn't my department, but I believe the issue here is that the translator uses `processDocuments()` to load non-HTML/XML data. This works in the client because the client's `processDocuments()` uses a hidden browser, which gives you a Document, and the text can be extracted via a virtual `pre` element. But translation-server's `processDocuments()` uses XMLHttpRequest's `responseType = 'document'`, which will return null for non-Document response (e.g., MARC data).

This could probably fixed by using `doGet()` instead in the translator. @simonster could say whether this is something that could be fixed easily in translation-server. Based on 92c698c7ef02f827fe52990f6f8b081b42c59779, it seems like using 'document' (instead of doing conditional parsing of `responseText`) might be done solely to allow for doc.defaultView, which is only used in three translators.

**Comment by Halibutt** [comment]
@dstillman, thank you! I didn't understand much of what you wrote above, but at least there is hope :)

**Comment by Halibutt** [comment]
So, @simonster & @dstillman , any chance anyone looks into it? How can I help?
